bundles/org.openhab.binding.zwavejs/src/main/java/org/openhab/binding/zwavejs/internal/conversion/BaseMetadata.java

```diff
@@ -0,0 +1,603 @@
+    }
+
+    /*
+     * Checks if the given map represents an RGB Color or Color Temperature map.
+     * The map should have either all the three keys "red", "green", and "blue",
+     * and/or one or two of the keys "warmWhite" and "coldWhite".
+     *
+     * @param map the map to check
+     *
+     * @return true if the map represents an RGB color map, false otherwise
+     */
+    private boolean isColorMap(Map<?, ?> map) {
+        return (map.containsKey("red") && map.containsKey("green") && map.containsKey("blue"))
+                || map.containsKey("warmWhite") || map.containsKey("coldWhite");
+    }
+
+    protected String itemTypeFromMetadata(MetadataType type, @Nullable Object value, int commandClass,
+            @Nullable Map<String, String> optionList) {
+        type = correctedType(type, value, commandClass, optionList);
+
+        switch (type) {
+            case NUMBER:
+                Unit<?> unit = this.unit;
+                if (unit != null) {
+                    String dimension = UnitUtils.getDimensionName(unit);
+                    if (dimension == null) {
+                        logger.warn("Node {}. Could not parse '{}' as a unit, fallback to 'Number' itemType", nodeId,
+                                unitSymbol);
+                        return CoreItemFactory.NUMBER;
+                    }
+                    return CoreItemFactory.NUMBER + ":" + dimension;
+                }
+
+                return CoreItemFactory.NUMBER;
+            case BOOLEAN:
+                // switch (or contact ?)
+                return CoreItemFactory.SWITCH;
+            case COLOR:
+                return CoreItemFactory.COLOR;
+            case STRING:
+            case STRING_ARRAY:
+                return CoreItemFactory.STRING;
+            default:
+                logger.error(
+                        "Node {}. Unable to determine item type based on metadata.type: {}, fallback to 'String' please file a bug report",
+                        nodeId, type);
+                return CoreItemFactory.STRING;
+        }
+    }
+
+    protected @Nullable StateDescriptionFragment createStatePattern(boolean writeable, @Nullable Integer min,
+            @Nullable Long max, @Nullable Integer step, @Nullable Object value) {
+        String pattern = "";
+        String itemTypeSplitted[] = itemType.split(":");
+        switch (itemTypeSplitted[0]) {
+            case CoreItemFactory.NUMBER:
+                String numberFormat = "%1d";
+                if (value instanceof Double) {
+                    // TODO: how to properly determine the decimals?
+                    numberFormat = "%.2f";
+                }
+                if (itemTypeSplitted.length > 1) {
+                    pattern = numberFormat + " %unit%";
+                } else {
+                    pattern = numberFormat;
+                }
+                break;
+            case CoreItemFactory.DIMMER:
+                pattern = "%1d %%";
```

Comment:
I do not remember, what is the difference between %1d and %d ?

Reply:
Can't find a good answer. %1d is better for percentage AFAICT as it does not allow negatives.